tests/seed_selection/seed_selection_suite_test.go

```diff
@@ -0,0 +1,373 @@
+// Copyright DataStax, Inc.
+// Please see the included license file for details.
+
+package seed_selection
+
+import (
+	"fmt"
+	"testing"
+	"strings"
+	"sort"
+	"regexp"
+	"encoding/json"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+
+	ginkgo_util "github.com/datastax/cass-operator/mage/ginkgo"
+	"github.com/datastax/cass-operator/mage/kubectl"
+)
+
+var (
+	testName            = "Seed Selection"
+	namespace           = "test-seed-selection"
+	dcName              = "dc1"
+	dcYaml              = "../testdata/default-three-rack-three-node-dc.yaml"
+	operatorYaml        = "../testdata/operator.yaml"
+	dcResource          = fmt.Sprintf("CassandraDatacenter/%s", dcName)
+	dcLabel             = fmt.Sprintf("cassandra.datastax.com/datacenter=%s", dcName)
+	ns                  = ginkgo_util.NewWrapper(testName, namespace)
+	defaultResources    = []string{
+		"../../operator/deploy/role.yaml",
+		"../../operator/deploy/role_binding.yaml",
+		"../../operator/deploy/service_account.yaml",
+		"../../operator/deploy/crds/cassandra.datastax.com_cassandradatacenters_crd.yaml",
+	}
+)
+
+func TestLifecycle(t *testing.T) {
+	AfterSuite(func() {
+		logPath := fmt.Sprintf("%s/aftersuite", ns.LogDir)
+		kubectl.DumpAllLogs(logPath).ExecV()
+
+		fmt.Printf("\n\tPost-run logs dumped at: %s\n\n", logPath)
+		_ = ns.Terminate()
+	})
+
+	RegisterFailHandler(Fail)
+	RunSpecs(t, testName)
+}
+
+func retrievePodNames(ns ginkgo_util.NsWrapper, dcName string) []string {
+	json := "jsonpath={.items[*].metadata.name}"
+	k := kubectl.Get("pods").
+		WithFlag("selector", fmt.Sprintf("cassandra.datastax.com/datacenter=%s", dcName)).
+		FormatOutput(json)
+
+	output := ns.OutputPanic(k)
+	podNames := strings.Split(output, " ")
+	sort.Sort(sort.StringSlice(podNames))
+
+	return podNames
+}
+
+type Node struct {
+	Name    string
+	Rack    string
+	Ready   bool
+	Seed    bool
+	Started bool
+	IP      string
+	Ordinal int
+}
+
+type DatacenterInfo struct {
+	Size      int
+	RackNames []string
+	Nodes     []Node
+}
+
+func retrieveNodes() []Node {
```

Comment:
At some point, we should look into moving some of this logic into our testing framework and share it between tests.